### PR TITLE
mpv: 0.37.0 -> 0.38.0 

### LINKED ISSUE #305164: Update request: mpv 0.37.0 → 0.38.0
- Package name: [mpv](https://mpv.io/)
- Latest released version: [0.38.0](https://github.com/mpv-player/mpv/releases/tag/v0.38.0)

- Current version on the unstable channel: [0.37.0](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/video/mpv/default.nix#L266)
- Current version on the stable/release channel: [0.36.0](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/pkgs/applications/video/mpv/default.nix#L252)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@AndersonTorres @fpletz @globin @Ma27 @tadeokondrak

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

closes https://github.com/NixOS/nixpkgs/issues/305164

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
hopefully closes https://github.com/NixOS/nixpkgs/issues/305164

**Comment by davidkna** [comment]
@AndersonTorres The darwin patch shouldn't be needed anymore. This also means the package requires `darwin.codesign` instead of `rcodesign` again.

**Comment by AndersonTorres** [comment]
1. I will then delete the patch
2. Still needs Darwin testers, since I am not one of them :)

File: pkgs/applications/video/mpv/default.nix [hunk 1]
```diff
@@ -154,18 +160,22 @@ in stdenv'.mkDerivation (finalAttrs: {
...
-  ++ lib.optionals waylandSupport [ wayland-scanner ];
+  ++ lib.optionals waylandSupport [
+    wayland-scanner
+  ]
+  ++ lib.optionals swiftSupport [
+    swift
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    xcbuild.xcrun
+  ];
```
Comment (davidkna) [inline pkgs/applications/video/mpv/default.nix:178]: `darwin.sigtool` is required again without the patch. It was forwarded via `inherit (darwin) sigtool;` in `all-packages.nix`:
```suggestion
  ++ lib.optionals stdenv.isDarwin [
    sigtool
    xcbuild.xcrun
  ];
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -33223,7 +33223,7 @@ with pkgs;
   shaka-packager = callPackage ../applications/video/shaka-packager { };
 
   # Wraps without triggering a rebuild
-  wrapMpv = callPackage ../applications/video/mpv/wrapper.nix { };
+  inherit (mpv-unwrapped) wrapMpv;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:33226]: Why not move this to aliases.nix?

File: pkgs/applications/video/mpv/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/default.nix:1]: Does require mpv a specific version of this library? If so, this change is kinda pointless.

File: pkgs/applications/video/mpv/default.nix [hunk 3]
```diff
@@ -1,103 +1,134 @@
...
+  vapoursynth,
+  vulkan-headers,
+  vulkan-loader,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+  xcbuild,
+  zimg,
 
-, waylandSupport ? stdenv.isLinux
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/default.nix:25]: I think the ordering here served a purpose and keeping it is worthwile the effort.

File: pkgs/applications/video/mpv/default.nix [hunk 4]
```diff
@@ -178,68 +193,181 @@ in stdenv'.mkDerivation (finalAttrs: {
...
-    ++ lib.optionals waylandSupport     [ wayland wayland-protocols libxkbcommon ]
-    ++ lib.optionals x11Support         [ libX11 libXext libGLU libGL libXxf86vm libXrandr libXpresent ]
-    ++ lib.optionals xineramaSupport    [ libXinerama ]
-    ++ lib.optionals xvSupport          [ libXv ]
-    ++ lib.optionals zimgSupport        [ zimg ]
-    ++ lib.optionals stdenv.isLinux     [ nv-codec-headers ]
-    ++ lib.optionals stdenv.isDarwin    [ libiconv ]
-    ++ lib.optionals stdenv.isDarwin    [ Accelerate CoreFoundation Cocoa CoreAudio MediaPlayer VideoToolbox ]
-    ++ lib.optionals (stdenv.isDarwin && swiftSupport) [ AVFoundation CoreMedia ];
+  ]
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/default.nix:191]: Those conditions are extremely long when not written into one line. IMO we shouldn't do that.

File: pkgs/applications/video/mpv/default.nix [hunk 5]
```diff
@@ -178,68 +193,181 @@ in stdenv'.mkDerivation (finalAttrs: {
...
+    substituteInPlace meson.build \
+      --replace-fail "conf_data.set_quoted('CONFIGURATION', configuration)" \
+                     "conf_data.set_quoted('CONFIGURATION', '<ommited>')"
+    ''
+    # A trick to patchShebang everything except mpv_identify.sh
+    ''pushd TOOLS''
+    ''mv mpv_identify.sh mpv_identify''
+    ''patchShebangs *.py *.sh''
+    ''mv mpv_identify mpv_identify.sh''
+    ''popd''
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/default.nix:329]: This should become a multi line string again

File: pkgs/applications/video/mpv/default.nix [hunk 6]
```diff
@@ -178,68 +193,181 @@ in stdenv'.mkDerivation (finalAttrs: {
...
-    printf "NoDisplay=true\n" >> umpv.desktop
-    popd
-  '' + lib.optionalString stdenv.isDarwin ''
-    mkdir -p $out/Applications
-    cp -r mpv.app $out/Applications
-  '';
+    ''sed -e '/Icon=/ ! s|mpv|umpv|g; s|^Exec=.*|Exec=umpv %U|' \
+        mpv.desktop > umpv.desktop''
+    ''printf "NoDisplay=true\n" >> umpv.desktop''
+    ''popd''
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/default.nix:360]: Same here

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:33226]: `aliases.nix` triggers ofBorg banishing mechanisms.

File: pkgs/applications/video/mpv/default.nix
Comment (AndersonTorres) [reply pkgs/applications/video/mpv/default.nix:1]: What are you talking about?

File: pkgs/applications/video/mpv/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/applications/video/mpv/default.nix:25]: I don't believe so.
The flags are self-explanatory, and the code below exposes their purpose.

File: pkgs/applications/video/mpv/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/applications/video/mpv/default.nix:191]: I would prefer something like a map for these things.

File: pkgs/applications/video/mpv/default.nix (hunk 5 above)
Comment (AndersonTorres) [reply pkgs/applications/video/mpv/default.nix:329]: I have mixed feelings about this. The indentation of multiline strings is less than good.

**Comment by AndersonTorres** [comment]
What is happening that this expression is not building?

File: pkgs/applications/video/mpv/default.nix (hunk 4 above)
Comment (JohnRTitor) [reply pkgs/applications/video/mpv/default.nix:191]: `lib.optional` can be used for conditional single element?

Comment (AndersonTorres) [reply pkgs/applications/video/mpv/default.nix:191]: Yes.

**Comment by JohnRTitor** [comment]
> What is happening that this expression is not building?

If I had to guess it could be related to wrapping mpv? You could try moving it to `pkgs/by-name`.

**Comment by AndersonTorres** [comment]
`by-name` - related errors are catched by a specific 

Currently mpv uses that dreadful `darwin.apple_sdk.callPackage` instead of plain `callPackage`. It makes the migration impossible.

**Comment by AndersonTorres** [comment]
@ofborb build mpv-unwrapped

**Comment by eclairevoyant** [comment]
> Currently mpv uses that dreadful `darwin.apple_sdk.callPackage` instead of plain `callPackage`. It makes the migration impossible.

It can still be migrated, just changing the path to callPackage accordingly

**Comment by AndersonTorres** [comment]
> It can still be migrated, just changing the path to callPackage accordingly

Please teach me.
I have tried something similar with `tinycc` and failed miserably.

**Comment by AndersonTorres** [comment]
> It can still be migrated, just changing the path to callPackage accordingly

No, it does not. Saving my fingers:

https://github.com/NixOS/nixpkgs/pull/305509#issuecomment-2099663683

**Comment by eclairevoyant** [comment]
Yeah, I just tested and ended up on the same error.
https://github.com/NixOS/nixpkgs-check-by-name/blob/7d6d26de7b271498d003e239b2a62cf22966398f/src/nixpkgs_problem.rs#L273-L277

**Comment by marsam** [comment]
I'm not completely sure why it fails to eval, but you can use the following as a workaround:
```diff
diff --git i/pkgs/applications/video/mpv/default.nix w/pkgs/applications/video/mpv/default.nix
index 748c75d88595..8ab099be8767 100644
--- i/pkgs/applications/video/mpv/default.nix
+++ w/pkgs/applications/video/mpv/default.nix
@@ -11,6 +11,7 @@
   ninja,
   pkg-config,
   python3,
+  sigtool,
   ffmpeg,
   freefont_ttf,
   freetype,
@@ -210,7 +211,7 @@ stdenv'.mkDerivation (finalAttrs: {
       pkg-config
     ]
     ++ lib.optionals stdenv.isDarwin [
-      darwin.sigtool
+      sigtool
       xcbuild.xcrun
     ]
     ++ lib.optionals swiftSupport [ swift ]
diff --git i/pkgs/top-level/all-packages.nix w/pkgs/top-level/all-packages.nix
index 7e202325ea50..99e11b8ed7ee 100644
--- i/pkgs/top-level/all-packages.nix
+++ w/pkgs/top-level/all-packages.nix
@@ -33005,6 +33005,7 @@ with pkgs;
 
   mpv-unwrapped = darwin.apple_sdk_11_0.callPackage ../applications/video/mpv {
     stdenv = if stdenv.isDarwin then swiftPackages.stdenv else stdenv;
+    inherit (darwin) sigtool;
     inherit lua;
   }; 
```

**Comment by AndersonTorres** [comment]
Let's try.

**Comment by AndersonTorres** [comment]
OfBorg is now all-green.
Anything more to be done?

**Comment by marsam** [comment]
>Anything more to be done?

please mark it as broken on Darwin. I'll try to fix it later

**Comment by marsam** [comment]
LGTM, Thank you! 
But since it seems to break the package on Darwin, I think it should be merged after the 24.05 branch-off (planned to happen in a couple of days).

**Comment by AndersonTorres** [comment]
Another conflict? What is happening here?

**Comment by JohnRTitor** [comment]
Looks like #313271 is responsible.

**Comment by AndersonTorres** [comment]
Nope, it was some reformatting in a wrapper...

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -33034,15 +33034,14 @@ with pkgs;
     inherit lua;
   };
 
-  shaka-packager = callPackage ../applications/video/shaka-packager { };
+  # Wrap avoiding rebuild
+  mpv = mpv-unwrapped.wrapper { mpv = mpv-unwrapped; };
 
-  # Wraps without triggering a rebuild
-  wrapMpv = callPackage ../applications/video/mpv/wrapper.nix { };
```
Comment (Scrumplex) [inline pkgs/top-level/all-packages.nix:33040]: Removal of wrapMpv breaks `pkgs/by-name/sv/svp/mvp.nix`

```
       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "wrapMpv" at /home/scrumplex/.cache/nixpkgs-review/pr-304349/nixpkgs/pkgs/by-name/sv/svp/mpv.nix:3'
```

Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:33040]: Fixed.

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 304349` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>ki</li>
    <li>ki.dist</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
  </ul>
</details>
<details>
  <summary>105 packages built:</summary>
  <ul>
    <li>adl</li>
    <li>ani-cli</li>
    <li>anilibria-winmaclinux</li>
    <li>anime-downloader</li>
    <li>anime-downloader.dist</li>
    <li>anki</li>
    <li>anki.dist</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>castero</li>
    <li>castero.dist</li>
    <li>celluloid</li>
    <li>cplay-ng</li>
    <li>cplay-ng.dist</li>
    <li>curseradio</li>
    <li>curseradio.dist</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>delfin</li>
    <li>dmlive</li>
    <li>dra-cla</li>
    <li>ff2mpv</li>
    <li>ff2mpv-go</li>
    <li>flet-client-flutter</li>
    <li>flet-client-flutter.debug</li>
    <li>flet-client-flutter.pubcache</li>
    <li>gonic</li>
    <li>gtk-pipe-viewer</li>
    <li>gtk-pipe-viewer.devdoc</li>
    <li>haruna</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>hypnotix</li>
    <li>invidtui</li>
    <li>jellyfin-media-player</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>jftui</li>
    <li>kdePackages.mpvqt</li>
    <li>kdePackages.mpvqt.debug</li>
    <li>kdePackages.mpvqt.dev</li>
    <li>kdePackages.mpvqt.devtools</li>
    <li>kdePackages.plasmatube</li>
    <li>kdePackages.plasmatube.debug</li>
    <li>kdePackages.plasmatube.dev</li>
    <li>kdePackages.plasmatube.devtools</li>
    <li>kdePackages.tokodon</li>
    <li>kdePackages.tokodon.debug</li>
    <li>kdePackages.tokodon.dev</li>
    <li>kdePackages.tokodon.devtools</li>
    <li>klipperscreen</li>
    <li>libsForQt5.plasmatube</li>
    <li>libsForQt5.tokodon</li>
    <li>memento</li>
    <li>minitube</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mov-cli</li>
    <li>mov-cli.dist</li>
    <li>mpc-qt</li>
    <li>mpv</li>
    <li>mpv-unwrapped</li>
    <li>mpv-unwrapped.dev</li>
    <li>mpv-unwrapped.doc</li>
    <li>mpv-unwrapped.man</li>
    <li>mpvScripts.acompressor</li>
    <li>mpvScripts.autocrop</li>
    <li>mpvScripts.autodeint</li>
    <li>mpvScripts.autoload</li>
    <li>mpvScripts.inhibit-gnome</li>
    <li>mpvScripts.mpris</li>
    <li>mpvpaper</li>
    <li>photoqt</li>
    <li>pipe-viewer</li>
    <li>pipe-viewer.devdoc</li>
    <li>plex-media-player</li>
    <li>plex-mpv-shim</li>
    <li>plex-mpv-shim.dist</li>
    <li>python311Packages.flet</li>
    <li>python311Packages.flet-runtime</li>
    <li>python311Packages.flet-runtime.dist</li>
    <li>python311Packages.flet.dist</li>
    <li>python311Packages.mpv</li>
    <li>python311Packages.mpv.dist</li>
    <li>python312Packages.flet</li>
    <li>python312Packages.flet-runtime</li>
    <li>python312Packages.flet-runtime.dist</li>
    <li>python312Packages.flet.dist</li>
    <li>python312Packages.mpv</li>
    <li>python312Packages.mpv.dist</li>
    <li>qimgv</li>
    <li>radioboat</li>
    <li>somafm-cli</li>
    <li>spotube</li>
    <li>stremio</li>
    <li>subtitleedit</li>
    <li>supersonic</li>
    <li>supersonic-wayland</li>
    <li>svp</li>
    <li>tomato-c</li>
    <li>wtwitch</li>
    <li>youtube-tui</li>
    <li>ytfzf</li>
    <li>ytui-music</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
[ki.log](https://github.com/NixOS/nixpkgs/files/15408892/ki.log)
[sublime-music.log](https://github.com/NixOS/nixpkgs/files/15408895/sublime-music.log)

sublime-music might not be compatible with this mpv version. Root cause seems to be `ValueError: ('Invalid value for mpv parameter', -4, (<MpvHandle object at 0x7fffe988b1d0>, <mpv.LP_MpvNode object at 0x7fffe988bd10>, <mpv.LP_MpvNode object at 0x7fffe988bc80>))`

`ki` has many failed tests.

**Comment by AndersonTorres** [comment]
Maybe they should be updated to Git HEAD.

**Comment by AndersonTorres** [comment]
A cursory look at Sublime Music looks like its mpv code is not modified two years:

https://github.com/sublime-music/sublime-music/blob/master/sublime_music/players/mpv.py

Maybe it deserves some upstream update?

**Comment by AndersonTorres** [comment]
According to the small investigation I did, I will rule Sublime as broken.

Starting at

https://github.com/sublime-music/sublime-music/blob/0b4ba69a7ff7ad2dfcb0a264587921f323030766/tests/player_tests/mpv_tests.py#L28

It directs to

https://github.com/sublime-music/sublime-music/blob/0b4ba69a7ff7ad2dfcb0a264587921f323030766/sublime_music/players/mpv.py#L137

that goes to pythonPackages.mpv:

https://github.com/jaseg/python-mpv/blob/5bb298ad11b7ad7dbeeddfd87546c0d34b28397c/mpv.py#L1229

that throws the exception at

https://github.com/jaseg/python-mpv/blob/5bb298ad11b7ad7dbeeddfd87546c0d34b28397c/mpv.py#L142

**Comment by marsam** [comment]
I'm sorry for the delay. I've been awfully busy these days.

I managed to fix the build on Darwin: [0001-mpv-fix-build-on-darwin.patch](https://github.com/user-attachments/files/15746554/0001-mpv-fix-build-on-darwin.patch)
Would you mind adding that patch in this PR?   It's not blocking, so it might as well be in another PR

**Review by marsam — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Patch applied!

**Comment by AndersonTorres** [comment]
@Scrumplex I have patched the expression for sublime.  Can you please test it?

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 304349` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>ki</li>
    <li>ki.dist</li>
  </ul>
</details>
<details>
  <summary>107 packages built:</summary>
  <ul>
    <li>adl</li>
    <li>ani-cli</li>
    <li>anilibria-winmaclinux</li>
    <li>anime-downloader</li>
    <li>anime-downloader.dist</li>
    <li>anki</li>
    <li>anki.dist</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>castero</li>
    <li>castero.dist</li>
    <li>celluloid</li>
    <li>cplay-ng</li>
    <li>cplay-ng.dist</li>
    <li>curseradio</li>
    <li>curseradio.dist</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>delfin</li>
    <li>dmlive</li>
    <li>dra-cla</li>
    <li>ff2mpv</li>
    <li>ff2mpv-go</li>
    <li>flet-client-flutter</li>
    <li>flet-client-flutter.debug</li>
    <li>flet-client-flutter.pubcache</li>
    <li>gonic</li>
    <li>gtk-pipe-viewer</li>
    <li>gtk-pipe-viewer.devdoc</li>
    <li>haruna</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>hypnotix</li>
    <li>invidtui</li>
    <li>jellyfin-media-player</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>jftui</li>
    <li>kdePackages.mpvqt</li>
    <li>kdePackages.mpvqt.debug</li>
    <li>kdePackages.mpvqt.dev</li>
    <li>kdePackages.mpvqt.devtools</li>
    <li>kdePackages.plasmatube</li>
    <li>kdePackages.plasmatube.debug</li>
    <li>kdePackages.plasmatube.dev</li>
    <li>kdePackages.plasmatube.devtools</li>
    <li>kdePackages.tokodon</li>
    <li>kdePackages.tokodon.debug</li>
    <li>kdePackages.tokodon.dev</li>
    <li>kdePackages.tokodon.devtools</li>
    <li>klipperscreen</li>
    <li>libsForQt5.plasmatube</li>
    <li>libsForQt5.tokodon</li>
    <li>memento</li>
    <li>minitube</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mov-cli</li>
    <li>mov-cli.dist</li>
    <li>mpc-qt</li>
    <li>mpv</li>
    <li>mpv-unwrapped</li>
    <li>mpv-unwrapped.dev</li>
    <li>mpv-unwrapped.doc</li>
    <li>mpv-unwrapped.man</li>
    <li>mpvScripts.acompressor</li>
    <li>mpvScripts.autocrop</li>
    <li>mpvScripts.autodeint</li>
    <li>mpvScripts.autoload</li>
    <li>mpvScripts.inhibit-gnome</li>
    <li>mpvScripts.mpris</li>
    <li>mpvpaper</li>
    <li>photoqt</li>
    <li>pipe-viewer</li>
    <li>pipe-viewer.devdoc</li>
    <li>plex-media-player</li>
    <li>plex-mpv-shim</li>
    <li>plex-mpv-shim.dist</li>
    <li>python311Packages.flet</li>
    <li>python311Packages.flet-runtime</li>
    <li>python311Packages.flet-runtime.dist</li>
    <li>python311Packages.flet.dist</li>
    <li>python311Packages.mpv</li>
    <li>python311Packages.mpv.dist</li>
    <li>python312Packages.flet</li>
    <li>python312Packages.flet-runtime</li>
    <li>python312Packages.flet-runtime.dist</li>
    <li>python312Packages.flet.dist</li>
    <li>python312Packages.mpv</li>
    <li>python312Packages.mpv.dist</li>
    <li>qimgv</li>
    <li>radioboat</li>
    <li>somafm-cli</li>
    <li>spotube</li>
    <li>stremio</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
    <li>subtitleedit</li>
    <li>supersonic</li>
    <li>supersonic-wayland</li>
    <li>svp</li>
    <li>tomato-c</li>
    <li>wtwitch</li>
    <li>youtube-tui</li>
    <li>ytfzf</li>
    <li>ytui-music</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
Seems to build fine now!

**Review by marsam — APPROVED** [review]
LGTM. Thank you!

**Comment by teto** [comment]
shouldn't there be an alias/redirection instead of hitting:
```
    error: attribute 'wrapMpv' missing
       at /nix/store/1jnssw33bwg5cywr9arnd4pg02pdx4l3-source/modules/programs/mpv.nix:59:5:
           58|   else
           59|     pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = cfg.scripts; };
```

**Comment by sersorrel** [comment]
Some note about how to fix breakage resulting from this would have been nice :)

I came up with something like:

```diff
     programs.mpv = {
       enable = true;
-      package = pkgs.wrapMpv (pkgs.mpv-unwrapped.override {
-        ffmpeg = pkgs.ffmpeg-full;
-      }) {
+      package = (pkgs.mpv-unwrapped.wrapper {
+        mpv = pkgs.mpv-unwrapped.override {
+          ffmpeg = pkgs.ffmpeg-full;
+        };
+      }).override {
         scripts = [
           pkgs.mpvScripts.mpris
```

**Comment by AndersonTorres** [comment]
> shouldn't there be an alias/redirection instead of hitting:
> 
> ```
>     error: attribute 'wrapMpv' missing
>        at /nix/store/1jnssw33bwg5cywr9arnd4pg02pdx4l3-source/modules/programs/mpv.nix:59:5:
>            58|   else
>            59|     pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = cfg.scripts; };
> ```

The interface changed in the new release.
However why wasn't this catched by ofBorg? o.O

**Comment by eclairevoyant** [comment]
> The interface changed in the new release. Alias would not work

At least a `throw` would be ideal to point at the new interface

> why wasn't this catched by ofBorg

borgo doesn't test code in files that you didn't touch, unless you tell it to, or there's some package test, etc

**Comment by sersorrel** [comment]
there was a test, but it seems like it wasn't set up properly so ofborg could run it: https://github.com/NixOS/nixpkgs/pull/318588#issuecomment-2156779224

seems like there was little/no other usage of wrapMpv in nixpkgs itself, so nixpkgs-review would not have helped either

> The interface changed in the new release. Alias would not work

wouldn't something like this work? (completely untested)

```nix
wrapMpv = unwrapped: (pkgs.mpv-unwrapped.wrapper { mpv = unwrapped; }).override;
```

**Comment by AndersonTorres** [comment]
> borgo doesn't test code in files that you didn't touch, unless you tell it to, or there's some package test, etc

indeed this code was external, from home-manager.

**Comment by Profpatsch** [comment]
Hi, why didn’t you add a backwards-compat layer for wrapMpv? Or a deprecation period?

**Comment by AndersonTorres** [comment]
We did not ponder it at that time.

Besides, the discussions about, ehr, breakages on unstable are very hot:

https://discourse.nixos.org/t/can-we-please-stop-breaking-stuff-willy-nilly/48496

**Comment by winterqt** [comment]
How is "people are discussing how to handle breakages" an excuse for *breaking more things*? At least stick with the first reason of "nobody realized," which is at least logical.

**Pull request merged**